### PR TITLE
fix: blending compatible with sktime 0.7.0

### PR DESCRIPTION
**Pull request by TremaMiguel**
#### Describe the changes you've made

Reimplement EnsembleForecaster with sktime>=0.7.0

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/internal/ensemble.py [hunk 1]
```diff
@@ -1,140 +1,129 @@
...
+            pred_w = np.average(pred_forecasters, axis=1, weights=self.weights)
+            return pd.DataFrame(pred_w, index=pred_forecasters.index)
+
+    @property
+    def _set_weights_none(self):
+        self.weights = None
+
+    @property
+    def _set_weights_uniform(self):
+        self.weights = np.ones(len(self.forecasters))
```
Comment (Yard1) [inline pycaret/internal/ensemble.py:129]: Shouldn't those be methods? Is this how sktime does it?

Comment (TremaMiguel) [reply pycaret/internal/ensemble.py:129]: Sktime original implementation doesn't handle `voting` method, here I should be changing these to appropiate setter for `@property`

```python
@property
 def weights(self):
      return self.weights

@weights.setter
def weights(self, value):
    self.weights = value
```

and then call

```python
elif self.method in self._not_required_weights and self.weights:
        warnings.warn(
            "Unused 'weights' argument. When method='mean' or method='median', 'weights' argument is not provided. Setting weights to `None`"
        )
        self.weights = None
```

Comment (TremaMiguel) [reply pycaret/internal/ensemble.py]: @Yard1 I think we should open a PR to include `voting` in sktime original implementation, what are your thoughts on this ?

Comment (Yard1) [reply pycaret/internal/ensemble.py]: That's a great idea @TremaMiguel

**Review by ngupta23 — APPROVED** [review]

**Review by Yard1 — APPROVED** [review]
LGTM!

**Pull request merged**